### PR TITLE
Migrate to pyproject.toml

### LINKED ISSUE #288: Migrate project to pyproject.toml
Is there interest in contributions to migrate the project to pyproject.toml?

### PR DESCRIPTION
**Pull request by lovetox**
Closes  #288

### CODE REVIEW

**Comment by lovetox** [comment]
I ran the tests, it does not work on python 3.6, because setuptools on that python version does not support editable installs for projects with only pyproject.toml

I see the following options

- Drop python 3.6 compatibility
- Disable the editable install test
- Wait with the move to pyproject.toml

**Comment by cvzi** [comment]
I think, remove the editable install. I see no reason why the editable install would need to be tested, not sure why I added that in the first place.

**Comment by lovetox** [comment]
I found more problems, the short story is, setuptools 59 was just not far enough in the implementation for pyproject.toml.
I added a commit to drop python 3.6 compatibility.

I looked at the stats and run them for march 2024

 python 3.6 |  0.35% │  10,700

**Review by cvzi — COMMENTED** [review]
I am okay with removing support for Python 3.6.

@TahirJalilov What do you think?

File: emoji/__init__.py
[the remark this answers is not in the document]
Comment (lovetox) [reply emoji/__init__.py:1]: i added them back, once the review is finished i can squash the commit

**Comment by TahirJalilov** [comment]
Thank you for your PR!

**Pull request merged**